src/components/gridHOC.jsx

```diff
@@ -182,44 +195,82 @@ export default (Grid: StaticDatagrid) => {
                 {column.name}
               </Table.HeaderCell>
             )))}
+            {this.props.editable && this.props.editIndividualRows && (
+              <Table.HeaderCell />
+            )}
           </Table.Row>
         </Table.Header>
       );
     }
 
     buildTableFooter() {
       const {
+        formName,
+        fieldName,
         editable,
         editIndividualRows,
+        bulkEdit,
         startEditingContent,
         addContent,
+        removeMultiple,
         editButtonLabel,
+        bulkEditButtonLabel,
+        bulkDeleteButtonLabel,
+        selectAllButtonLabel,
+        selectAllFilteredButtonLabel,
+        unselectAllButtonLabel,
         addButtonLabel,
         exportable,
         exportButtonLabel,
         exportFileName,
         columnModel,
+        classes,
       } = this.props;
 
       const data = this.state.store.getData();
 
+      let columnSpan = this.colModel.get().length;
+      if (editable && editIndividualRows) {
+        columnSpan += 1;
+      }
+      if (editable && bulkEdit) {
+        columnSpan += 1;
+      }
+
       return (
         <Table.Footer fullWidth>
           <Table.Row>
-            <Table.HeaderCell colSpan={this.colModel.get().length + (editIndividualRows ? 1 : 0)}>
+            <Table.HeaderCell colSpan={columnSpan}>
               {editable && (
-                <div style={{ float: "left" }}>
+                <div className={classes.footerButtons}>
                   <EditControls
+                    data={data}
                     editIndividualRows={editIndividualRows}
+                    bulkEdit={bulkEdit}
                     startEditingContent={startEditingContent}
                     editButtonLabel={editButtonLabel}
+                    bulkEditButtonLabel={bulkEditButtonLabel}
+                    bulkDeleteButtonLabel={bulkDeleteButtonLabel}
+                    selectAllButtonLabel={selectAllButtonLabel}
+                    selectAllFilteredButtonLabel={selectAllFilteredButtonLabel}
+                    unselectAllButtonLabel={unselectAllButtonLabel}
                     addContent={addContent}
+                    removeMultiple={removeMultiple}
                     addButtonLabel={addButtonLabel}
+                    editHandler={this.editHandler}
+                    selectableData={this.paginationHandler.data}
+                    isFiltered={(
+                      !!this.searchHandler.tagsList.length || !!this.searchHandler.inputValue
+                    )}
```

Comment:
Need to figure out if there is a way to avoid the gridHOC from digging into plugins. Ideally, the HOC should only be aware of what plugins are being used and invoke them.

Reply:
> Need to figure out if there is a way to avoid the gridHOC from digging into plugins. Ideally, the HOC should only be aware of what plugins are being used and invoke them.

This change is not needed for now.